app/controllers/ops_controller/settings/analysis_profiles.rb

```diff
@@ -378,51 +379,21 @@ def ap_form_field_changed
   # Delete all selected or single displayed scanitemset(s)
   def ap_delete
     assert_privileges("ap_delete")
-    scanitemsets = []
-    if !params[:id] # showing a list
-      scanitemsets = find_checked_ids_with_rbac(ScanItemSet)
-      if scanitemsets.empty?
-        add_flash(_("No Analysis Profiles were selected for deletion"), :error)
+    @single_delete = true if params[:id]
+    scanitemsets = find_records_with_rbac(ScanItemSet, checked_or_params)
+    scanitemsets.each do |scan_item_set|
+      if scan_item_set.read_only
+        scanitemsets -= [scan_item_set]
+        add_flash(_("Default Analysis Profile \"%{name}\" can not be deleted") % {:name => scan_item_set.name}, :error)
       else
-        to_delete = []
-        scanitemsets.each do |s|
-          scan = ScanItemSet.find(s)
-          if scan.read_only
-            to_delete.push(s)
-            add_flash(_("Default Analysis Profile \"%{name}\" can not be deleted") % {:name => scan.name}, :error)
-          end
-        end
-        # deleting elements in temporary array, had to create temp array to hold id's to be delete, .each gets confused if i deleted them in above loop
-        to_delete.each do |a|
-          scanitemsets.delete(a)
-        end
-      end
-      @flash_error = true  if scanitemsets.empty?
-      scanitemsets.each do |id|
-        itemset = ScanItemSet.find(id)
-        mems = itemset.members
-        mems_to_delete = []
-        mems.each do |m|
-          mems_to_delete.push(m)
-        end
-        ap_deletescanitems(mems_to_delete)
-        # resetting flash_array to prevent from showing message from deleting each scanitem under scanitemset
-        @flash_array = []
-        itemset.remove_all_members
-      end
-      ap_process_scanitemsets(scanitemsets, "destroy")  unless scanitemsets.empty?
-    else # showing 1 scanitemset, delete it
-      if params[:id].nil? || ScanItemSet.find_by_id(params[:id]).nil?
-        add_flash(_("Analysis Profile no longer exists"), :error)
-      else
-        scanitemsets.push(params[:id])
-      end
-      @single_delete = true
-      ap_process_scanitemsets(scanitemsets, "destroy")  unless scanitemsets.empty?
-      if @flash_array.nil?
-        add_flash(_("The selected Analysis Profile was deleted"))
+        tmp_flash_array = @flash_array
+        ap_deletescanitems(scan_item_set.members)
+        @flash_array = tmp_flash_array
+        scan_item_set.remove_all_members
       end
     end
+    @flash_error = true if scanitemsets.empty?
```

Comment:
What happens when `@flash_error` is set to `false` rather than `nil` ? Could this be:
```ruby
@flash_error = scanitemsets.empty?
```

Reply:
thanks, I'll fix it now :+1: